File: workspace-arena/Clase5/tvSeries/src/main/xtend-gen/unq/edu/uis/tvSeries/domain/TvSeriesSystem.java
package unq.edu.uis.tvSeries.domain;

import java.util.List;
import java.util.function.Predicate;
import org.eclipse.xtend.lib.annotations.Accessors;
import org.eclipse.xtext.xbase.lib.Functions.Function1;
import org.eclipse.xtext.xbase.lib.IterableExtensions;
import org.eclipse.xtext.xbase.lib.Pure;
import org.uqbar.commons.model.annotations.Observable;
import unq.edu.uis.tvSeries.domain.Serie;

@Accessors
@Observable
@SuppressWarnings("all")
public class TvSeriesSystem {
  private List<Serie> series;
  
  public void agregarSerie(final Serie serie) {
    this.series.add(serie);
  }
  
  public void borrarSerie(final Serie serie) {
    this.series.remove(serie);
  }
  
  public Serie buscarSerie(final String nombre) {
    final Function1<Serie, Boolean> _function = new Function1<Serie, Boolean>() {
      public Boolean apply(final Serie it) {
        return Boolean.valueOf(it.getNombre().toLowerCase().contains(nombre.toLowerCase()));
      }
    };
    return IterableExtensions.<Serie>findFirst(this.series, _function);
  }
  
  public boolean eliminarSeriePorId(final Integer id) {
    final Predicate<Serie> _function = new Predicate<Serie>() {
      public boolean test(final Serie it) {
        int _id = it.getId();
        return (_id == (id).intValue());
      }
    };
    return this.series.removeIf(_function);
  }
  
  @Pure
  public List<Serie> getSeries() {
    return this.series;
  }
  
  public void setSeries(final List<Serie> series) {
    this.series = series;
  }
}

File: ejemplos/dominios/src/main/java/tvSeries/TvSeriesSystem.kt
package tvSeries

class TvSeriesSystem {
	val series = mutableListOf<Serie>()

	fun addSerie(serie: Serie?) {
		if (serie == null) return
		series.add(serie)
	}

	fun removeActorFromSerie(actor: Actor?, serie: Serie?) {
		if (actor == null || serie == null) return
		series.firstOrNull { it.actors.contains(actor) }?.removeActor(actor)
	}

	fun removeSerie(serie: Serie?) = series.remove(serie)

	fun searchSerie(text: String) =
			series.firstOrNull { it.name.toLowerCase().contains(text.toLowerCase()) }

	fun removeSerieById(id: Int) = series.removeIf { it.id == id }
}
